src/common/rbac/project/robot.go

```diff
@@ -0,0 +1,61 @@
+package project
+
+import "github.com/goharbor/harbor/src/common/rbac"
+
+// robotContext the context interface for the robot
+type robotContext interface {
+	// Index whether the robot is authenticated
+	IsAuthenticated() bool
+	// GetUsername returns the name of robot
+	GetUsername() string
+	// GetPolicy get the rbac policies from security context
+	GetPolicies() []*rbac.Policy
+}
+
+// robot implement the rbac.User interface for project robot account
+type robot struct {
```

Comment:
This struct can be moved to the package where robot security context sits, and the robotContext is not needed.  We can reuse the robot security context in the struct.